addons/test_mail_sms/tests/test_phone_blacklist.py

```diff
@@ -50,6 +50,33 @@ def test_phone_blacklist_internals(self):
             rec = self.env['mail.test.sms.bl'].search([('phone_sanitized_blacklisted', '=', True)])
             self.assertEqual(rec, self.env['mail.test.sms.bl'])
 
+    def test_phone_blacklist_unblacklisted(self):
+        """ This test check for scenario where user:
+            1. Blacklists a number  -> creating new active record)
+            2. Unblacklists it      -> making record unactive (archived)
+            3. Blacklists it again (by attempting to create new record)
+        Last step should just make existing record active again, and this test checks it.
+        """
+        phone_number = self.test_numbers[0]
+        bl_record = self.env['phone.blacklist'].sudo().create([{'number': phone_number, 'active': True}])
+        num_of_records = self.env['phone.blacklist'].with_context(active_test=False).search_count([])
+
+        bl_record.action_archive()
+        self.env['phone.blacklist'].flush_model()
```

Comment:
You should not have to flush the model manually, the query inside the create should do that. (so you won't need to add it at all if we go with `search_fetch`)

Reply:
yup, after switching to ORM it work fine